test/lib/ufe/testPointInstances.py

```diff
@@ -0,0 +1,113 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import mayaUtils
+import usdUtils
+
+from mayaUsd import ufe as mayaUsdUfe
+
+from pxr import UsdGeom
+
+import ufe
+
+import unittest
+
+
+class PointInstancesTestCase(unittest.TestCase):
+    '''
+    Tests that the UFE path and scene item interfaces work as expected when
+    working with USD point instances.
+    '''
+
+    _pluginsLoaded = False
+
+    @classmethod
+    def setUpClass(cls):
+        if not cls._pluginsLoaded:
+            cls._pluginsLoaded = mayaUtils.isMayaUsdPluginLoaded()
+
+    def setUp(self):
+        self.assertTrue(self._pluginsLoaded)
+
+        mayaUtils.openPointInstancesGrid14Scene()
+
+    def testPointInstances(self):
+        # Create a UFE path to a PointInstancer prim with an instanceIndex on
+        # the end. This path uniquely identifies a specific point instance.
+        ufePath = ufe.Path([
+            mayaUtils.createUfePathSegment('|UsdProxy|UsdProxyShape'),
+            usdUtils.createUfePathSegment('/PointInstancerGrid/PointInstancer/7')])
+        ufeItem = ufe.Hierarchy.createItem(ufePath)
+
+        # Verify that we turned the path into a scene item correctly and that
+        # we can access the PointInstancer prim.
+        self.assertEqual(ufeItem.nodeType(), 'PointInstancer')
+
+        ufeAttrs = ufe.Attributes.attributes(ufeItem)
+        self.assertTrue(ufeAttrs.hasAttribute(UsdGeom.Tokens.positions))
+
+        # XXX: Note that UsdSceneItem currently has no wrapping to Python, so
+        # we are not able to verify that we can query the scene item directly
+        # for its instance index. We can exercise the Utils functions though.
+
+        # XXX: Another note, would be nice here to be able to do
+        # ufePath.string() or str(ufePath) to produce this path string instead.
+        ufePathString = ','.join([str(segment) for segment in ufePath.segments])
+
+        prim = mayaUsdUfe.ufePathToPrim(ufePathString)
+        self.assertTrue(prim)
+        self.assertTrue(prim.IsA(UsdGeom.PointInstancer))
+
+        instanceIndex = mayaUsdUfe.ufePathToInstanceIndex(ufePathString)
+        self.assertEqual(instanceIndex, 7)
+
+        # Tests for the usdPathToUfePathSegment() utility function.
+        ufePathSegmentString = mayaUsdUfe.usdPathToUfePathSegment(prim.GetPrimPath())
+        self.assertEqual(ufePathSegmentString, '/PointInstancerGrid/PointInstancer')
+
+        ufePathSegmentString = mayaUsdUfe.usdPathToUfePathSegment(prim.GetPrimPath(), 7)
+        self.assertEqual(ufePathSegmentString, '/PointInstancerGrid/PointInstancer/7')
+
+        # XXX: Similar to the issue above, Ufe::Path::string() yields a string
```

Comment:
Is the comment still relevant?  You're not creating scene items in lines 88-91.

Reply:
Addressed in 1d344b1db3abc06e9b58396c0e181b49ece3e979 and 32299f7a276e9e0040b5fd06890515509012f204.